app/controllers/application_controller/buttons.rb

```diff
@@ -794,7 +794,7 @@ def button_valid?(button_hash = @edit[:new])
       add_flash(_('URL can be opened only by buttons for a single entity'), :error)
     end
 
-    if !button_hash[:dialog_id].empty? && button_hash[:display_for] != 'single'
```

Comment:
I think you need the negation.

If dialog is given and button is to be displayed for list of items, then flash.